File: s3-inf-ppo/laboratoria/lab10/java/src/Notebook/Entries/WatchedMovie.java
package Notebook.Entries;

import Notebook.Interfaces.NotebookEntry;
import Notebook.Traits.Slugger;

public class WatchedMovie implements NotebookEntry, Slugger {

    private String title;

    public WatchedMovie(String title) {
        this.title = title;
    }

    @Override
    public String getSlug() {
        return getSlug(title);
    }

}

File: s3-inf-ppo/laboratoria/lab10/php/src/Entries/WatchedMovie.php
<?php

namespace PPO\Notebook\Entries;

use PPO\Notebook\Interfaces\NotebookEntry;
use PPO\Notebook\Traits\Slugger;

class WatchedMovie implements NotebookEntry {

	use Slugger;

	private $title;

	public function __construct(string $title) {
		$this->title = $title;
	}

	protected function getSlugBase(): string {
		return $this->title;
	}

}
